pkg/run/session.go

```diff
@@ -0,0 +1,68 @@
+package run
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	vcluster "github.com/loft-sh/vcluster/cmd/vclusterctl/cmd"
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/flags"
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/log"
+	"github.com/mitchellh/go-ps"
+)
+
+type Session struct {
+	name string
+}
+
+func (s *Session) Connect(namespace string) error {
+	subProcArgs := append(os.Args, "--disable-session", "--allow-k8s-context="+s.name)
+	fmt.Println("subProcArgs:", subProcArgs)
+
+	connect := vcluster.ConnectCmd{
+		GlobalFlags: &flags.GlobalFlags{
+			Silent:    true,
+			Namespace: namespace,
+		},
+		Log:                   log.GetInstance(),
+		BackgroundProxy:       false, // must be false to avoid creating the docker container
+		KubeConfigContextName: s.name,
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-c
+
+		thisProc := os.Getpid()
+		allProcesses, err := ps.Processes()
+
+		if err != nil {
+			return
+		}
+
+		for _, proc := range allProcesses {
+			if proc.PPid() == thisProc {
+				// ok it's a child process, notify it
+				pp, err := os.FindProcess(proc.Pid())
```

Comment:
❔ Here I just didn't quite get what we are doing, could you please clarify it a bit?

`Pid`s are unique for each process, correct? So, if we already have a process `proc` and just checked that its parent process's ID is `thisProc`, why do we also need to search for `pp` here `pp, err := os.FindProcess(proc.Pid())`? Because if process IDs are unique, wouldn't be the `pp` process returned by `os.FindProcess(proc.Pid())`  the same process as `proc`?

❔ BTW, how should the `pp` abbreviation/variable name be deciphered here?

Reply:
The loop is to find it (an `int`), and `os.FindProcess()` is to obtain the process object (a `struct` representing the process).

Reply:
Ah, ` proc` is of type `ps.Process` and `pp`, returned by `os.FindProcess(` is of type `*os.Process`, so they are different types and we actually need to get the object of type `*os.Process` in the end?

Reply:
Oh. Good catch. Let me revisit it.

Reply:
No. `proc` is of `ps.Process`. What we need to use is `*os.Process`.
`ps.Process` belongs to the `go-ps` package which is for process enumeration.

Reply:
Yes, that's what I wrote. Thanks for confirming it.